gap/PolygonalComplexes/constructors.gi

```diff
@@ -459,13 +459,121 @@ __SIMPLICIAL_IntSetConstructor("VerticesInFaces", __SIMPLICIAL_AllTypes,
 ##  UmbrellaDescriptorOfSurface . . . . . compute the umbrella descriptor
 ##
 ##
-InstallMethod( UmbrellaDescriptorOfSurface, 
-    "for a simplicial surface", [IsSimplicialSurface], 
-    function( surf) 
+
+BindGlobal("__SIMPLICIAL_OrientateUmbrellaDescriptor",
+    function(surface, commonDescr)
+        local numFaces, firstBoundIndex, i, faces, face, vertexAdjencies,
+            vertices, vertexPaths, findPredecessorBfs, vertexPredsMap, vertexRotationCycleMap,
+            orientedDescr, pair, vertex, predVertex, predRotationCycle, rotationCycle,
+        # BFS related variables:
+            queue, visited, referenceVertex;
+
+        numFaces := Length(Faces(surface));
+
+        if Length(commonDescr) = 1 or numFaces = 1 then
+            return commonDescr;
+        fi;
+
+        # Find the first bound index in commonDescr
+        for i in [1..Length(commonDescr)] do
+            if not IsBound(commonDescr[i]) then
+                continue;
+            fi;
+
+            firstBoundIndex := i;
+            break;
+        od;
+
+        # For each vertex find all vertex adjancies as [vertex, face] pairs
+        vertexAdjencies := List(commonDescr, x -> []);
+        for face in [1..numFaces] do
+            vertices := VerticesOfFace(surface, face);
```

Comment:
It could be that the face list is also not bound, i.e. use better `for face in Faces(surf) do`.